radixdlt-java/src/main/java/com/radixdlt/client/core/crypto/ECKeyPairGenerator.java

```diff
@@ -19,23 +20,28 @@
 import org.bouncycastle.crypto.params.ECDomainParameters;
 import org.bouncycastle.jce.provider.BouncyCastleProvider;
 import org.bouncycastle.jce.spec.ECParameterSpec;
+import org.radix.crypto.Hash;
 
 public final class ECKeyPairGenerator {
-	private static final Map<Integer, ECDomainParameters> DOMAINS;
+    private static final Map<Integer, ECDomainParameters> DOMAINS = new HashMap<>();
 
-	static {
-		if (AndroidUtil.isAndroidRuntime()) {
-			Security.removeProvider(BouncyCastleProvider.PROVIDER_NAME);
-		}
-		Security.insertProviderAt(new BouncyCastleProvider(), 1);
-		DOMAINS = Stream.of(256).collect(Collectors.toMap(Integer::new, (bits) -> {
-			final X9ECParameters curve =
-				bits == 256
-					? CustomNamedCurves.getByName("secp" + bits + "k1")
-					: SECNamedCurves.getByName("secp" + bits + "k1");
-			return new ECDomainParameters(curve.getCurve(), curve.getG(), curve.getN(), curve.getH());
-		}));
-	}
+    static {
+        install();
+    }
+
+    static void install() {
+        if (AndroidUtil.isAndroidRuntime()) {
```

Comment:
Can we add a check or flag here so that we don't call `install()` multiple times?

Reply:
Good catch! I'll look into this tomorrow (there myst be a way to check what version of bouncy castle is currently loaded in `Security`).

Reply:
Fixed!

Having worked a bit closer with this particular code, I must say I don't fully see the benefit of involving `stream`'s, with all the whistles and bells that comes with it, only to add a single `Integer` mapping to a `HashMap`:

```java
DOMAINS.putAll(Stream.of(256)
        .collect(Collectors.toMap(Integer::new, (bits) -> {
                final X9ECParameters curve = bits == 256
                        ? CustomNamedCurves.getByName("secp" + bits + "k1")
                        : SECNamedCurves.getByName("secp" + bits + "k1");
                return new ECDomainParameters(curve.getCurve(), curve.getG(), curve.getN(), curve.getH());
        })));
```

vs.

```java
X9ECParameters curve = CustomNamedCurves.getByName("secp" + bits + "k1");
DOMAINS.put(256, new ECDomainParameters(curve.getCurve(), curve.getG(), curve.getN(), curve.getH());
```